articles/hdinsight/hbase/apache-hbase-rest-sdk.md

```diff
@@ -1,187 +1,187 @@
-
-## Select data
-
-To read data from an HBase table, pass the table name and row key to the `GetCellsAsync` method to return the `CellSet`.
-
-```csharp
-var key = "fifth_element";
-
-var cells = await client.GetCellsAsync("RestSDKTable", key);
-// Get the first value from the row.
-Console.WriteLine(Encoding.UTF8.GetString(cells.rows[0].values[0].data));
-// Get the value for t1:title
-Console.WriteLine(Encoding.UTF8.GetString(cells.rows[0].values
-    .Find(c => Encoding.UTF8.GetString(c.column) == "t1:title").data));
-// With the previous insert, it should yield: "The Fifth Element"
-```
-
-In this case, the code returns only the first matching row, as there should only be one row for a unique key. The returned value is changed into `string` format from the `byte[]` array. You can also convert the value to other types, such as an integer for the movie's release date:
-
-```csharp
-var releaseDateField = cells.rows[0].values
-    .Find(c => Encoding.UTF8.GetString(c.column) == "t1:release_date");
-int releaseDate = 0;
-
-if (releaseDateField != null)
-{
-    releaseDate = Convert.ToInt32(Encoding.UTF8.GetString(releaseDateField.data));
-}
-Console.WriteLine(releaseDate);
-// Should return 1997
-```
-
-## Scan over rows
-
-HBase uses `scan` to retrieve one or more rows. This example requests multiple rows in batches of 10, and retrieves data whose key values are between 25 and 35. After retrieving all rows, delete the scanner to clean up resources.
-
-```csharp
-var tableName = "mytablename";
-
-// Assume the table has integer keys and we want data between keys 25 and 35
-var scanSettings = new Scanner()
-{
-	batch = 10,
-	startRow = BitConverter.GetBytes(25),
-	endRow = BitConverter.GetBytes(35)
-};
-RequestOptions scanOptions = RequestOptions.GetDefaultOptions();
-scanOptions.AlternativeEndpoint = "hbaserest0/";
-ScannerInformation scannerInfo = null;
-try
-{
-    scannerInfo = await client.CreateScannerAsync(tableName, scanSettings, scanOptions);
-    CellSet next = null;
-    while ((next = client.ScannerGetNextAsync(scannerInfo, scanOptions).Result) != null)
-    {
-	    foreach (var row in next.rows)
-        {
-    	    // ... read the rows
-        }
-    }
-}
-finally
-{
-    if (scannerInfo != null)
-    {
-        await client.DeleteScannerAsync(tableName, scannerInfo, scanOptions);
-    }
-}
-```
-
-## Next steps
-
-* [Get started with an Apache HBase example in HDInsight](apache-hbase-tutorial-get-started-linux.md)
-* Build an end-to-end application with [Analyze real-time Twitter sentiment with HBase](../hdinsight-hbase-analyze-twitter-sentiment.md)
+---
+title: Use the HBase .NET SDK - Azure HDInsight 
+description: Use the HBase .NET SDK to create and delete tables, and to read and write data.
+services: hdinsight
+author: ashishthaps
+ms.reviewer: jasonh
+
+ms.service: hdinsight
+ms.custom: hdinsightactive
+ms.topic: conceptual
+ms.date: 12/13/2017
+ms.author: ashishth
+---
+# Use the Apache HBase .NET SDK

```

Comment:
Wondering if we can clarify this H1. Is the SDK titled like this? or could we say `Use the .Net SDK for Apache HBase`. Apache didn't make the SDK I assume.

Reply:
updated